src/boot/efi/boot.c

```diff
@@ -612,105 +677,222 @@ static EFI_STATUS reboot_into_firmware(void) {
         assert_not_reached();
 }
 
-static bool menu_run(
-                Config *config,
-                ConfigEntry **chosen_entry,
-                char16_t *loaded_image_path) {
+static char16_t *create_padded_line(const char16_t *title_show, size_t line_width) {
+        assert(title_show);
+
+    size_t j, padding = (line_width - MIN(strlen16(title_show), line_width)) / 2;
+    char16_t *line = xnew(char16_t, line_width + 1);
+    for (j = 0; j < padding; j++)
+        line[j] = ' ';
+    for (size_t k = 0; title_show[k] != '\0' && j < line_width; j++, k++)
+        line[j] = title_show[k];
+    for (; j < line_width; j++)
+        line[j] = ' ';
+    line[line_width] = '\0';
```

Comment:
While you're touching all this: This kind of thing should be doable with something like `xasprintf(%*.*s);` etc. I'd prefer going away from this kind of string manipulations.

Reply:
Should this happen here or in another PR?

Reply:
Whatever you think works best for you. It's okay for this PR to contain refactors, just don't squash it into a single commit.

Reply:
Alright, I'll split it into multiple commits 👍🏻